test/Test/IO/Tinfoil/AEAD/AESGCM/Foreign.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Test.IO.Tinfoil.AEAD.AESGCM.Foreign where
+
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import qualified Data.Text as T
+
+import           Disorder.Core.IO (testIO)
+import           Disorder.Core.Property (failWith)
+import           Disorder.Core.Tripping (tripping)
+
+import           P
+
+import           System.IO
+
+import           Test.QuickCheck
+import           Test.Tinfoil.Arbitrary ()
+
+import           Tinfoil.AEAD.AESGCM.Data
+import           Tinfoil.AEAD.AESGCM.Foreign
+import           Tinfoil.Data.Key
+import           Tinfoil.Internal.Sodium
+
+import           X.Control.Monad.Trans.Either (runEitherT)
+
+testSodium :: (SodiumInitMarker -> Property) -> Property
+testSodium p =
+  testIO $ (runEitherT initialiseSodium) >>= \x -> pure $ case x of
+    Right mark -> p mark
+    Left err -> failWith . T.pack $ show err
+
+prop_aes256Gcm_encrypt :: Cleartext -> AssociatedData -> SymmetricKey -> GcmIv -> Property
+prop_aes256Gcm_encrypt clear ad sk iv =
+  testSodium $ \m ->
+    let
+      ct = aes256GcmEncrypt m clear ad sk iv
+    in
+    (BS.isInfixOf (unCleartext clear) (unAuthenticatedCiphertext ct)) === False
+
```

Comment:
Tiny minor bikeshedding thing. Pretty sure, the outer parens aren't needed there. Just thought I'd mention it. Fixing it is totally up to you.